src/aind_ophys_utils/motion_border_utils.py

```diff
@@ -0,0 +1,149 @@
+""" Utils to get motion borders """
+from collections import namedtuple
+
+import numpy as np
+import pandas as pd
+from pathlib import Path
+
+# see comment in motion_border_from_max_frame_shift for an explanation
+# of the difference between these two structures
+
+MaxFrameShift = namedtuple("MaxFrameShift", ["left", "right", "up", "down"])
+
+MotionBorder = namedtuple(
+    "MotionBorder", ["top", "bottom", "left_side", "right_side"]
+)
+
+
+def get_max_correction_values(
+    x_series: pd.Series, y_series: pd.Series, max_shift: float = 30.0
+) -> MaxFrameShift:
+    """
+    Gets the max correction values in the cardinal directions from a series
+    of correction values in the x and y directions
+
+    Parameters
+    ----------
+    x_series: pd.Series:
+        A series of movements in the x direction
+    y_series: pd.Series:
+        A series of movements in the y direction
+    max_shift: float
+        Maximum shift to allow when considering motion correction. Any
+        larger shifts are considered outliers (only absolute value matters).
+
+    For deprecated implementation see:
+    allensdk.internal.brain_observatory.roi_filter_utils.calculate_max_border
+
+    Returns
+    -------
+    MaxFrameShift
+        A named tuple containing the maximum correction values found during
+        motion correction workflow step. Saved with the following direction
+        order [left, right, up, down].
+
+    """
+    # take abs of max shift as we are considering both positive and negative
+    # directions
+    max_shift = abs(max_shift)
+
+    # filter based out analomies based on maximum_shift
+    x_no_outliers = x_series[
+        (x_series >= -max_shift) & (x_series <= max_shift)
+    ]
+    y_no_outliers = y_series[
+        (y_series >= -max_shift) & (y_series <= max_shift)
+    ]
+    # calculate max border shifts
+    right_shift = -1 * x_no_outliers.min()
+    left_shift = x_no_outliers.max()
+    down_shift = -1 * y_no_outliers.min()
+    up_shift = y_no_outliers.max()
+
+    max_shift = MaxFrameShift(
+        left=left_shift, right=right_shift, up=up_shift, down=down_shift
+    )
+
+    # check if all exist
+    if np.any(np.isnan(np.array(max_shift))):
+        raise ValueError(
+            "One or more motion correction shifts "
+            "was found to be Nan, max shift found: "
+            f"{max_shift}, with max_shift {max_shift}"
+        )
+
+    return max_shift
+
+
+def get_max_correction_from_file(
```

Comment:
let's have `input_csv` be a `input_df` pd.DataFrame.  We should avoid file I/O as much as possible.